src/network.js

```diff
@@ -62,8 +62,16 @@ export class BlockstackNetwork {
   }
 
   coerceAddress(address: string) {
-    const addressHash = bitcoinjs.address.fromBase58Check(address).hash
-    return bitcoinjs.address.toBase58Check(addressHash, this.layer1.pubKeyHash)
+    const { hash, version } = bitcoinjs.address.fromBase58Check(address)
```

Comment:
Lol, I have a similar snippet of code in `cli-blockstack`.  Glad this is getting added to blockstack.js!

Do you want to throw an exception of the `version` isn't either `scriptHash` or `pubKeyHash`?

Reply:
Sure, it now throws if it cannot recognize the version number